src/Interpreters/Cache/FileCache.cpp

```diff
@@ -783,6 +783,17 @@ bool FileCache::tryReserve(
     ProfileEventTimeIncrement<Microseconds> watch(ProfileEvents::FilesystemCacheReserveMicroseconds);
 
     assertInitialized();
+
+    /// A logical race on cache_is_being_resized is still possible,
+    /// in this case we will try to lock cache with timeout, this is ok, timeout is small
+    /// and as resizing of cache can take a long time then this small chance of a race is
+    /// ok compared to the number of cases this check will help.
+    if (cache_is_being_resized.load())
```

Comment:
maybe use `std::memory_order_relaxed` for all store/load if it's just approx/optimization to avoid trying to take lock